workflow-pacta-cop.Rmd

```diff
@@ -0,0 +1,516 @@
+
+
+```
+
+Now the second peer project is run with the aggregation at the peer group level. 
+
+```{r echo = FALSE}
+
+project_name <- paste0(project_code, "_peers")
+twodii_internal <- FALSE
+
+# TRUE or FALSE: TRUE means that the code is running on a 2dii laptop with dropbox connection
+#####################################################################
+### ONLY FOR EXTERNAL PROJECTS (twodii_internal <- FALSE):
+# Variables must exist for internal projects
+project_location_ext <- "C:/Users/clare/Desktop/ExternalTest"
+data_location_ext <- "C:/Users/clare/Desktop/TestData"
+#####################################################################
+
+create_project_folder(project_name, twodii_internal, project_location_ext)
+set_project_paths(project_name, twodii_internal, project_location_ext)
+copy_files(project_name)
+set_project_parameters(file.path(par_file_path,paste0("ProjectParameters.yml")))
+analysis_inputs_path <- set_analysis_inputs_path(twodii_internal, data_location_ext, dataprep_timestamp)
+
+inc_meta_portfolio <- FALSE
+write_csv(portfolio_peers, path = file.path(raw_input_path, paste0(project_name, "_Input.csv")))
+
+```
+
+Add the portfolio files to the folder `{r} file.path(raw_input_path)`.
+This file should have:
+- investor.name as the peer group name (banks, pensionfund etc)
+- portfolio.name also as the peer group name
+
+```{r echo = FALSE}
+# These run the above project. Will error out if the portfolio file is not provided in the input path
+# For more information on errors that arise, open the workflow-offline.Rmd file
+
+source("2_project_input_analysis.R")
+
+source("3_run_analysis.R")
+
+
+file.copy(file.path(results_path, "Equity_results_portfolio.rda"), file.path(data_path, paste0(project_code,"_peers_equity_results_portfolio.rda")), overwrite = T)
+
+file.copy(file.path(results_path, "Bonds_results_portfolio.rda"), file.path(data_path, paste0(project_code,"_peers_bonds_results_portfolio.rda")), overwrite = T)
+
+```
+
+### Run the QA rmd code
+
+This is a detailed look into the data being used in this analysis. 
+This file is saved in the results folder here: `{r} results_path`
+
+
+```{r echo = FALSE}
+
+ render(file.path("pacta_data_qa.Rmd"),
+         output_dir = results_path,
+         clean = FALSE,
+         quiet = TRUE,
+         params = list(
+            project_code = project_code,
+            project_location_ext = project_location_ext,
+            twodii_internal = twodii_internal
+         )
+         )
+
+
+```
+
+
+### Checking the missing isins
+
+Using the peer group files, it is necessary to check for missing isins. This needs to be obtained and the pacta-data repository updated if necessary. 
+
+Additional missing isins are appended to the file data/isins_missing_bbg_data.csv. This identifies all unique missing isins across all projects.  
+
+This should be analysed using Bloomberg to obtain any new missing isins and these be added to the financial data, before running this again. 
```

Comment:
This step has always been unclear to me, and frankly still is now. It seems like this would require a user to create a PR any time something changes `data/isins_missing_bbg_data.csv`? If the goal is to continually expand a list of ISINs we've found "in the wild" but that are unknown in our data, that's great, but I think there's probably a much better, more realistic way of doing that... I think quite possibly the `PACTA_analysis` repo is **not** the right place for it.

Sorry, I know that here you're just trying to document the current process as it is.

Reply:
Issue has been created